code/graphics/render.cpp

```diff
@@ -814,12 +814,26 @@ void gr_string(float sx, float sy, const char* s, int resize_mode, size_t in_len
 		GR_DEBUG_SCOPE("Render TTF string");
 
 		auto path = beginDrawing(resize_mode);
-		path->translate(sx, sy);
 
 		auto nvgFont = static_cast<NVGFont*>(currentFont);
 
+		float scale_factor = Font_Scale_Factor;
+		if (!nvgFont->getScaleBehavior()) {
+			scale_factor = 1.0f;
+		}
+
+		float originalSize = nvgFont->getSize();
+		float scaledSize = originalSize * scale_factor;
+
+		// Calculate the offset to center the text
+		float offsetX = 0.0f;
+		// Not sure if we should do this.. kinda depends on if the text is drawn at the top of the screen or the bottom
+		float offsetY = (scaledSize - originalSize) * 0.5f;
```

Comment:
Given that the text is set to ALIGN_TOP a few lines later, IMO the scaling should reflect that (as in, that the upper left corner of the text does not move when the text scales).
Rather than manually offset the position, it's probably better to let the text library handle that, and if necessary, make the text alignment configurable for stuff like scripting or maybe even hudgauges.
In the long run, this anchor needs to set properly (both horizontally and vertically) by whatever is drawing the text based on the context in which it's drawn to get proper anchor behaviour everywhere.

Reply:
Comments updated per our discussion on Discord